pkg/istio-agent/bootstrap_stream.go

```diff
@@ -37,29 +38,8 @@ type bootstrapDiscoveryStream struct {
 }
 
 // Send refers to a request from the xDS proxy.
+// This should be no called.
 func (b *bootstrapDiscoveryStream) Send(resp *discovery.DiscoveryResponse) error {
```

Comment:
@ramaraochavali seems broken from https://github.com/istio/istio/pull/30921

Reply:
Sorry. Did not understand. Can you please explain more?

Reply:
Because the `bootstrapDiscoveryStream` is a fake downstream of envoy, normally it will be called in `forwardToEnvoy`.
but we skipped for non-envoy type in `forwardToEnvoy`. So it can never run into this handler.

Reply:
I has to to have a special condition like` && resp.TypeUrl != v3.WorkloadType `

Reply:
That is an option, but still need to NACK/ACK.

The current way will not need to do ack/nack again.